etc/search-core.api.md

```diff
@@ -263,6 +263,31 @@ export interface Endpoints {
     verticalSearch?: string;
 }
 
+// @public
+export class EndpointsProvider {
+    // Warning: (ae-forgotten-export) The symbol "Environment" needs to be exported by the entry point index.d.ts
+    // Warning: (ae-forgotten-export) The symbol "CloudRegion" needs to be exported by the entry point index.d.ts
+    constructor(environment?: Environment, cloudRegion?: CloudRegion, overrideEndpoints?: Endpoints);

```

Comment:
the risk with requiring the config object is that this class would depend on many required properties which aren't relevant to this class. an alternative would be have the constructor take in a single object with environment and cloudRegion properties. that way if we were to update the config to have environment and cloudRegion properties, the config could be passed in and typescript would allow it based on duck typing

Reply:
I agree with Connor. I'll remove the overrideEndpoints and callers can implement their own override if that sounds good to people.

Reply:
My suggestion then would be to move the relevant endpoint, region, and environment properties to an interface that the config object extends.  That way different sub components don't have to know about the whole config object, but users only need to know about one and can just pass it around.